src/Microsoft.Diagnostics.EventFlow.Inputs.EventFlowTelemetryProcessor/EventFlowTelemetryProcessor.cs

```diff
@@ -0,0 +1,461 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Validation;
+using Microsoft.Diagnostics.EventFlow;
+using Microsoft.Diagnostics.EventFlow.Inputs;
+
+namespace Microsoft.Diagnostics.EventFlow.ApplicationInsights
+{
+    public class EventFlowTelemetryProcessor : ITelemetryProcessor
+    {
+        private static readonly LogLevel[] ToLogLevel = new LogLevel[]
+        {
+            LogLevel.Verbose,
+            LogLevel.Informational,
+            LogLevel.Warning,
+            LogLevel.Error,
+            LogLevel.Critical
+        };
+        public static readonly string TelemetryTypeProperty = "TelemetryType";
+
+        private ITelemetryProcessor next;
+        private IObserver<EventData> input;
+        private IHealthReporter healthReporter;
+
+        public EventFlowTelemetryProcessor(ITelemetryProcessor next)
+        {
+            Requires.NotNull(next, nameof(next));
+
+            this.next = next;
+        }
+
+        public DiagnosticPipeline Pipeline
+        {
+            set
+            {
+                Requires.NotNull(value, nameof(value));
+
+                this.input = value.Inputs.OfType<IItemWithLabels>()
+                    .Where(i => i.Labels.ContainsKey(ApplicationInsightsInputFactory.ApplicationInsightsInputTag))
+                    .Cast<IObserver<EventData>>().FirstOrDefault();
+                this.healthReporter = value.HealthReporter;
+            }
+        }
+
+        public void Process(ITelemetry item)
+        {
+            var currentInput = this.input;
+
+            if (currentInput == null)
+            {
+                this.next.Process(item);
+                return;
+            }
+
+            try
+            {
+                item.Sanitize();
+
+                var eventData = new EventData();
+                eventData.Timestamp = item.Timestamp;
+                eventData.ProviderName = "EventFlow-ApplicationInsightsInput";
+
+                var eventPayload = eventData.Payload;
+
+                var request = item as RequestTelemetry;
+                if (request != null)
```

Comment:
Use `if (item is EventArgs request) {}`. This way you can get rid of goto in favor of `} else if () {`